a10_octavia/cmd/vthunder_heartbeat_udp.py

```diff
@@ -75,47 +58,36 @@ def update(self, key, ip, port):
             self.sock = socket.socket(ai_family, socket.SOCK_DGRAM)
             self.sock.settimeout(1)
             self.sock.bind(self.sockaddr)
-            if cfg.CONF.a10_health_manager.sock_rlimit > 0:
-                rlimit = cfg.CONF.a10_health_manager.sock_rlimit
+            if CONF.a10_health_manager.sock_rlimit > 0:
+                rlimit = CONF.a10_health_manager.sock_rlimit
                 LOG.info("setting sock rlimit to %s", rlimit)
                 self.sock.setsockopt(socket.SOL_SOCKET, socket.SO_RCVBUF,
                                      rlimit)
             break  # just used the first addr getaddrinfo finds
         if self.sock is None:
-            raise exceptions.NetworkConfig("unable to find suitable socket")
-
-    def dorecv(self, *args, **kw):
-        """Waits for a UDP heart beat to be sent.
-
-        :return: Returns the unwrapped payload and addr that sent the
-                 heartbeat. The format of the obj from the UDP sender
-                 can be seen below. Note that listener_1 has no pools
-                 and listener_4 has no members.
-
-        """
-        (data, srcaddr) = self.sock.recvfrom(UDP_MAX_SIZE)
-        ip, port = srcaddr
-        LOG.warning('Received packet from %s', ip)
-        # get record id of first vthunder from srcaddr
-        record_id = self.vthunder_repo.get_vthunder_from_src_addr(db_api.get_session(), ip)
-
-        if record_id:
-            last_udp_update = datetime.datetime.utcnow()
-            self.vthunder_repo.update(db_api.get_session(), record_id,
-                                      last_udp_update=last_udp_update)
-
-        return srcaddr[0]
+            raise exceptions.NetworkConfig("Unable to find suitable socket")
 
     def check(self):
+        """Wait and obtain the source address of UDP packet and updates its time in
+           vThunder repositories.
+        """
         try:
-            srcaddr = self.dorecv()
+            (data, srcaddr) = self.sock.recvfrom(UDP_MAX_SIZE)
```

Comment:
Why's this wrapped in a tuple? Why not just `data, srcaddr = self.sock.recvfrom(UDP_MAX_SIZE)`? Just below you use `ip, port = scraddr` without wrapping it